Add a template for CSS functional notation (#28953)
mdn/content: files/en-us/mdn/writing_guidelines/page_structures/page_types/css_function_page_template/index.md

---
title: CSS function page template
slug: MDN/Writing_guidelines/Page_structures/Page_types/CSS_function_page_template
page-type: mdn-writing-guide
browser-compat: css.functions.NameOfTheFunction
---

{{MDNSidebar}}

> **Note:** _Remove this note block before publishing._
>
> ---
>
> **Page front matter:**
>
> The front matter at the top of the page is used to define "page metadata".

> > The values should be updated appropriately for the particular function.

Reviewer: optional edit, but helps prevent errors
```suggestion
> The values should be updated appropriately for the particular function. Note the presence (or absence) of parenthesis.
```
alternatively, add "with parenthesis" or "with no parenthesis" in the title, slug, and browser-compat lines.

This section is for interactive examples added using the `\{{EmbedInteractiveExample}}` macro. You create these examples in the [mdn/interactive-examples repository](https://github.com/mdn/interactive-examples/blob/main/CONTRIBUTING.md). See the [Interactive examples](/en-US/docs/MDN/Writing_guidelines/Page_structures/Code_examples#interactive_examples) section in our _Writing guidelines_ for more information.

## Syntax

Include a CSS code block showing the main syntax forms, including the parameters that can be given. Only include the function itself, not a complete declaration in which it occurs. Lines should not end in semicolons: this should emphasize that we are not showing complete valid CSS code here, just the syntax form.

You should show all the invocation patterns that the function can take, and include comments naming parameters, which should match the parameters listed in the "Parameters" section.

The comment showing each invocation pattern should be followed by exactly one empty line.

For example:

```css
/* Without a fallback */

/* var( <custom-property-name> ) */
var(--custom-prop)

/* With fallback */

/* var( <custom-property-name> , ) */
var(--custom-prop,)

> /* var( <custom-property-name> , <declaration-value> ) */

Author: I don't really know how prescriptive we need to be in general.

I have 2 main concerns:
- people should be able to cross-reference between this code block, the "Values" list, and the formal syntax. So we should if at all possible use the same names for things in all three places.
- we should use the code block as a place to _show_ the syntax, not to _describe_ the syntax.

> /* var( <custom-property-name> , <declaration-value> ) */

Reviewer 2: Agree, it can be a guideline instead of a requirement and authors and reviewers can take a call based on the context.

var(--custom-prop, initial)
var(--custom-prop, #FF0000)
var(--my-background, linear-gradient(transparent, aqua), pink)
var(--custom-prop, var(--default-value))
var(--custom-prop, var(--default-value, red))
```

### Parameters

Give the parameters that the function can accept, as a {{htmlelement("dl")}}.
Include one term and definition for each parameter.

- `<custom-property-name>`
  - : Include parameter's data type, its description, and what it represents.
- `<declaration-value>`
  - : Include a description of the parameter, its data type, and what it represents.

### Return value

Describe the value returned by the function.

## Description

> This is an optional section to include a description of the function and explain how it works. Use this section to explain related terms and add use cases for the function.

Reviewer: ```suggestion
Include a description of the function and explain how it works. Use this section to explain related terms and add use cases for the function.
```
Prefer to have this semi-required.

Author: I'm happy to recommend it but this edit makes it required, not semi-required. At the moment many of our pages don't have description sections, and if we make it mandatory we are likely to get low-value edits just to tick boxes in the template.

Reviewer: Or just "We highly recommend you include a description..."

Author: I think if it is optional, we should say it is optional. It's just clearer that way. I've pushed an update along the lines of Dipika's suggestion.

## Formal syntax

Not all functions have formal syntax: if a function doesn't, omit this whole section.

> `\{CSSSyntax}}`

Reviewer: ```suggestion
`\{{CSSSyntax}}`
```

_To use this macro, remove the backticks and backslash in the markdown file._

## Examples

Note that we use the plural "Examples" even if the page only contains one example.

### Add a descriptive heading

> Each example must have an H3 heading (`###`) naming the example. The heading should be descriptive of what the example is doing. For example, "A simple example" does not say anything about the example and therefore, not a good heading. The heading should be concise. For a longer description, use the paragraph after the heading.

Reviewer 2: 1. I always thought that in addition to making an H3 heading mandatory in "Examples", we also had some guidance to include a few words about the example. It might make more sense now if we get rid of the language H4 headers as well as "Result". We need to convey that the H3 should be followed by a description of the example scenario/setup or what we're trying to demo. And ideally, after the result frame, explain the output as a result of what we did in the example codes.

2. I realize that this would need to be updated across all the template files that have the "Examples" section and I can take that up we agree on the general idea to include that instruction.

3. [CSS property page template](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Page_types/CSS_property_page_template#add_a_descriptive_heading), just as an example, also has a "Note" block (I remember you added that across templates :) ). Are we going to add that here?

Author: >     3. [CSS property page template](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Page_types/CSS_property_page_template#add_a_descriptive_heading), just as an example, also has a "Note" block (I remember you added that across templates :) ). Are we going to add that here?

I've added this.

> 1. I always thought that in addition to making an H3 heading mandatory in "Examples", we also had some guidance to include a few words about the example. It might make more sense now if we get rid of the language H4 headers as well as "Result". We need to convey that the H3 should be followed by a description of the example scenario/setup or what we're trying to demo. And ideally, after the result frame, explain the output as a result of what we did in the example codes.
> 
>     2. I realize that this would need to be updated across all the template files that have the "Examples" section and I can take that up we agree on the general idea to include that instruction.

Can we split proposals for how to change the way we examples across all pages off from this request to add a template for a specific page type? This seems really out of scope for my PR here.

Reviewer 2: > Can we split proposals for how to change the way we examples across all pages off from this request to add a template for a specific page type? This seems really out of scope for my PR here.

Fair enough, better to discuss this separately

You should show all the invocation patterns that the function can take, and include comments naming parameters, which should match the parameters listed in the "Parameters" section.

The comment showing each invocation pattern should be followed by exactly one empty line.

For example:

```css
/* Without a fallback */

/* var( <custom-property-name> ) */
var(--custom-prop)

/* With fallback */

/* var( <custom-property-name> , ) */
var(--custom-prop,)

/* var( <custom-property-name> , <declaration-value> ) */
var(--custom-prop, initial)
var(--custom-prop, #FF0000)
var(--my-background, linear-gradient(transparent, aqua), pink)
var(--custom-prop, var(--default-value))
var(--custom-prop, var(--default-value, red))
```

### Parameters

Give the parameters that the function can accept, as a {{htmlelement("dl")}}.
Include one term and definition for each parameter.

- `<custom-property-name>`

>   - : Include a description of the parameter, its data type, and what it represents.

Reviewer 3: We mostly mention data type first.
```suggestion
  - : Include parameter's data type, its description, and what it represents.
```
